File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/UpdateHealthPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record UpdateHealthPacket(float health, short hunger, float saturation) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeFloat(health);
        os.writeShort(hunger);
        os.writeFloat(saturation);
    }

    @Override
    public byte getId() {
        return 0x06;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/UpdateHealthPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class UpdateHealthPacket(val health: Float, val hunger: Short, val saturation: Float) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeFloat(health)
        os.writeShort(hunger.toInt())
        os.writeFloat(saturation)
    }

    override fun getId(): Byte {
        return 0x06
    }
}
